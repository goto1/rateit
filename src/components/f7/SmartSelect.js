import React from "react";
import PropTypes from "prop-types";
import capitalize from "lodash/capitalize";
import head from "lodash/head";

const SmartSelect = ({
  name,
  options,
  selected = [],
  multiple = false,
  searchbarPlaceholder,
  onChange
}) => {
  const selectedOptions = multiple === false ? head(selected) : selected;
  return (
    <li>
      <a
        className="item-link smart-select"
        data-searchbar="true"
        data-searchbar-placeholder={searchbarPlaceholder}
      >
        <select
          value={selectedOptions}
          name={name}
          onChange={onChange}
          multiple={multiple}
        >
          {options.map(option =>
            <option
              key={option.id}
              value={option.id}
              data-display-as={option.abbreviation}
            >
              {option.name}
            </option>
          )}
        </select>
        <div className="item-content">
          <div className="item-inner">
            <div className="item-title">
              {capitalize(name)}
            </div>
            <div className="item-after" />
          </div>
        </div>
      </a>
    </li>
  );
};

// const SmartSelect = ({
//   name,
//   options,
//   selected = [],
//   multiple = false,
//   searchbarPlaceholder,
//   onChange
// }) =>
//   <li>
//     <a
//       className="item-link smart-select"
//       data-searchbar="true"
//       data-searchbar-placeholder={searchbarPlaceholder}
//     >
//       <select
//         value={selected}
//         name={name}
//         onChange={onChange}
//         multiple={multiple}
//       >
//         {options.map(option =>
//           <option
//             key={option.id}
//             value={option.id}
//             data-display-as={option.abbreviation}
//           >
//             {option.name}
//           </option>
//         )}
//       </select>
//       <div className="item-content">
//         <div className="item-inner">
//           <div className="item-title">
//             {capitalize(name)}
//           </div>
//           <div className="item-after" />
//         </div>
//       </div>
//     </a>
//   </li>;

SmartSelect.propTypes = {
  name: PropTypes.string.isRequired,
  options: PropTypes.array.isRequired,
  multiple: PropTypes.bool,
  selected: PropTypes.array,
  searchbarPlaceholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default SmartSelect;
