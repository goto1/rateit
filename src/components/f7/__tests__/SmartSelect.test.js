import React from "react";
import SmartSelect from "../SmartSelect";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";

const props = {
  name: "schools",
  options: [
    {
      id: "bruHy",
      name: "New Jersey Institute of Technology, Newark",
      abbreviation: "NJIT"
    },
    {
      id: "5y7fh",
      name: "Rutgers University, New Brunswick",
      abbreviation: "RU"
    }
  ],
  selected: ["bruHy"],
  searchbarPlaceholder: "Search for schools...",
  onChange: jest.fn()
};

describe("SmartSelect component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SmartSelect {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls a function with selection changes", () => {
    const { onChange } = props;
    const wrapper = shallow(<SmartSelect {...props} />);

    wrapper.find("select").simulate("change");
    expect(onChange.mock.calls.length).toBe(1);
  });
});
