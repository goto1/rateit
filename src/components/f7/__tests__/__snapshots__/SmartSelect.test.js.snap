// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartSelect component renders correctly 1`] = `


<li>
  <a
    className="item-link smart-select"
    data-searchbar="true"
    data-searchbar-placeholder="Search for schools..."
  >
    <select
      multiple={true}
      name="schools"
      onChange={[Function]}
      value={
        Array [
          "bruHy",
        ]
      }
    >
      <option
        data-display-as="NJIT"
        value="bruHy"
      >
        New Jersey Institute of Technology, Newark
      </option>
      <option
        data-display-as="RU"
        value="5y7fh"
      >
        Rutgers University, New Brunswick
      </option>
    </select>
    <div
      className="item-content"
    >
      <div
        className="item-inner"
      >
        <div
          className="item-title"
        >
          Schools
        </div>
        <div
          className="item-after"
        />
      </div>
    </div>
  </a>
</li>
`;
